src/exception.c

```diff
@@ -139,10 +139,21 @@ void exception_initialize(void)
     }
     if (cpu_features->fast_ipi) {
         reg_clr(SYS_IMP_APL_PMCR0, PMCR0_IACT | PMCR0_IMODE_MASK);
-        reg_clr(SYS_IMP_APL_UPMCR0, UPMCR0_IMODE_MASK);
         msr(SYS_IMP_APL_IPI_SR_EL1, IPI_SR_PENDING);
     }
 
+    switch (cpu_features->uncore_version) {
+        case UNCORE_V1:
+            reg_clr(SYS_IMP_APL_UPMCR0, UPMCR0_IMODE_T8015);
+            break;
+        case UNCORE_V2:
+            reg_clr(SYS_IMP_APL_UPMCR0, UPMCR0_IMODE_T8020);
+            break;
+        default: // In boot CPU's EL3 cpu_features is not initialized yet but it is 0 on machines
```

Comment:
`case UNCORE_NONE` and then we should get a compiler warning if we ever add _V3 and forget to change it here